.github/workflows/ci.yml

```diff
@@ -5,47 +5,42 @@ on:
     branches:
       - master
   pull_request:
-    branches:
-      - "*"
 
 jobs:
   test:
     name: Test
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
-        with:
-          toolchain: stable
-      - uses: actions-rs/cargo@v1
-        with:
-          command: test
+      - uses: dtolnay/rust-toolchain@nightly
+      - run: cargo test
 
   format:
     name: Format
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
+      - uses: dtolnay/rust-toolchain@nightly
         with:
-          toolchain: stable
           components: rustfmt
-      - uses: actions-rs/cargo@v1
-        with:
-          command: fmt
-          args: -- --check
+      - run: cargo fmt -- --check
 
   clippy:
     name: Clippy
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
+      - uses: dtolnay/rust-toolchain@nightly
         with:
-          toolchain: stable
           components: clippy
-          args: -- -D warnings
-      - uses: actions-rs/cargo@v1
+      - run: cargo clippy --all-targets
```

Comment:
Can you add `-- -D warnings` option?

Reply:
i've added `#![deny(clippy::all)]` to `lib.rs` instead. This has the same effect, but with the advantage of making the behaviour consistent between CI and local dev

Reply:
i see what you mean. the 'non-snake-case' lint is a compiler warning, not a clippy warning. I've updated the catch this too.

This is now configured in the root of each target.

The question now is, is the benefit of having the compiler flag inline in the code worth the tradeoff of having to configure it in three places? I think it is, but what's your preference?